src/lib.rs

```diff
@@ -41,6 +45,48 @@ use structopt::StructOpt;
 ///
 /// - `authors`: Author names, taken from usual environment variables (i.e.
 ///   those which are also used by Cargo and git)
+///
+/// The template author can define their own placeholders in their
+/// `cargo-generate.toml` file. This looks like the following:
+///
+/// [placeholders]
+/// my-placeholder = { type = "string", prompt = "Hello?", choices = ["hello", "world"], default = "hello", regex = "*" }
+/// use-serde = { type = "bool", prompt = "Add serde support?", default = false }
+///
+/// The user of the template will then be asked the the question in "prompt", and must accept the
+/// default value (if provided) or enter a custom value which will be checked against "choices" (if
+/// provided) and regex (if provided).
+///
+/// The placeholder map supports the following keys:
+///
+/// `type` [required]: Must be "string" or "bool"
```

Comment:
those brackets in `[required]` hints cargo doc to create a link to a field of a struct, but here we don't reference a field from a struct so it's required to either mask the brackets or remove / replace them with something else.

Reply:
thanks, I'll fix it